packages/server-wallet/src/evm-validator.ts

```diff
@@ -18,7 +18,7 @@ const bytecodeCache: Record<string, string | undefined> = {};
 export const validateTransitionWithEVM = async (
   from: State,
   to: State,
-  tx: Objection.Transaction
+  txOrKnex: TransactionOrKnex
```

Comment:
The type `Objection.Transaction` was intended to help prevent a deadlock bug: `validateTransitionWithEVM` should _only_ be run from within a `lockApp` transaction, and so the type here means the compiler can help prevent deadlocks.

From a test, we can pass `undefined as any` for the value of `tx`.

Reply:
Ok -- that probably deserves a code comment to explain. And it might have to be `knex as any` now?

Reply:
4d7af33